src/SFA.DAS.Employer.PR.Web/Controllers/Requests/ChangeNameController.cs

```diff
@@ -0,0 +1,83 @@
+{
+    public const string UserEmailDoesNotMatchRequestShutterPageViewPath = "~/Views/Requests/UserEmailDoesNotMatchRequestShutterPage.cshtml";
+    public const string RequestsChangeNameViewPath = "~/Views/Requests/CreateServiceAccountChangeName.cshtml";
+
+    [Authorize]
+    [HttpGet]
+    [Route("{requestId:guid}/createaccount/changename", Name = RouteNames.CreateAccountChangeName)]
+    public async Task<IActionResult> Index(Guid requestId, CancellationToken cancellationToken)
+    {
+        GetPermissionRequestResponse permissionRequest = await _outerApiClient.GetPermissionRequest(requestId, cancellationToken);
+
+        if (User.GetEmail() != permissionRequest.EmployerContactEmail) return View(UserEmailDoesNotMatchRequestShutterPageViewPath);
+
+        var sessionModel = _sessionService.Get<AccountCreationSessionModel>();
+        SetNamesInSessionModel(sessionModel, permissionRequest);
+
+        var backLink = Url.RouteUrl(RouteNames.CreateAccountCheckDetails, new { requestId });
+        EmployerAccountNamesModel vm = GetChangeNameViewModel(permissionRequest.EmployerContactFirstName!, permissionRequest.EmployerContactLastName!, backLink!);
+
+        return View(RequestsChangeNameViewPath, vm);
+    }
+
+    [Authorize]
+    [HttpPost]
+    [Route("{requestId:guid}/createaccount/changename", Name = RouteNames.CreateAccountChangeName)]
+    public IActionResult Index(Guid requestId, EmployerAccountNamesSubmitModel submitModel, CancellationToken cancellationToken)
+    {
+        var result = _validator.Validate(submitModel);
+        if (!result.IsValid)
+        {
+            EmployerAccountNamesModel model = GetChangeNameViewModel(submitModel.EmployerContactFirstName!,
+                submitModel.EmployerContactLastName!,
+                Url.RouteUrl(RouteNames.CreateAccountCheckDetails, new { requestId })!);
+            result.AddToModelState(ModelState);
+            return View(RequestsChangeNameViewPath, model);
+        }
+
+        var sessionModel = new AccountCreationSessionModel
+        {
+            FirstName = submitModel.EmployerContactFirstName,
+            LastName = submitModel.EmployerContactLastName
+        };
+        _sessionService.Set(sessionModel);
+
+        return RedirectToRoute(RouteNames.CreateAccountCheckDetails, new { requestId });
+    }
+
+    private static void SetNamesInSessionModel(AccountCreationSessionModel? sessionModel, GetPermissionRequestResponse? permissionRequest)
+    {
+        if (sessionModel != null && permissionRequest != null)
+        {
+            permissionRequest.EmployerContactFirstName = sessionModel.FirstName;
+            permissionRequest.EmployerContactLastName = sessionModel.LastName;
+        }
+    }
+
+
+    private static EmployerAccountNamesModel GetChangeNameViewModel(string firstName, string lastName, string backLink)
```

Comment:
EmployerAccountNamesModel -> EmployerAccountNamesViewModel

Reply:
By the way it is Employer User Name

Reply:
Ok i've reworked the complexities of the models to use a base class, calledm EmployerUserNamesBase :)

Reply:
There is no need for SubmitModel as it is same as ViewModel. Also The ViewModel needs to be in its own file. You can keep the base class, but I would not bother, just copy the two fields across in both the ViewModels. The ViewModel name is incorrect, it should be ChangeNameViewModel to match controller

Reply:
ok I've removed the base class and removed the submitViewModel, as you've suggested.